migration-roadmap/badge.md

```diff
@@ -0,0 +1,337 @@
+- `--mod-badge-subtle-background-color-brown`
+- `--mod-badge-subtle-background-color-celery`
+- `--mod-badge-subtle-background-color-chartreuse`
+- `--mod-badge-subtle-background-color-cinnamon`
+- `--mod-badge-subtle-background-color-cyan`
+- `--mod-badge-subtle-background-color-default`
+- `--mod-badge-subtle-background-color-fuchsia`
+- `--mod-badge-subtle-background-color-gray`
+- `--mod-badge-subtle-background-color-green`
+- `--mod-badge-subtle-background-color-indigo`
+- `--mod-badge-subtle-background-color-informative`
+- `--mod-badge-subtle-background-color-magenta`
+- `--mod-badge-subtle-background-color-negative`
+- `--mod-badge-subtle-background-color-notice`
+- `--mod-badge-subtle-background-color-orange`
+- `--mod-badge-subtle-background-color-pink`
+- `--mod-badge-subtle-background-color-positive`
+- `--mod-badge-subtle-background-color-purple`
+- `--mod-badge-subtle-background-color-red`
+- `--mod-badge-subtle-background-color-seafoam`
+- `--mod-badge-subtle-background-color-silver`
+- `--mod-badge-subtle-background-color-turquoise`
+- `--mod-badge-subtle-background-color-yellow`
+- `--mod-badge-subtle-label-icon-color`
+- `--mod-badge-workflow-icon-size`
+
+</details>
+
+### SWC
+
+<details>
+<summary>Attributes</summary>
+
+- `variant` - Badge variant (accent, neutral, informative, positive, negative, notice, fuchsia, indigo, magenta, purple, seafoam, yellow, gray, red, orange, chartreuse, celery, green, cyan, blue)
+- `fixed` - Fixed positioning (inline-start, inline-end, block-start, block-end)
+- `size` - Badge size (s, m, l, xl)
+
+</details>
+
+<details>
+<summary>Slots</summary>
+
+- Default slot - Text label of the badge
+- `icon` - Optional icon that appears to the left of the label
+
+</details>
+
+## Comparison
+
+### Visual comparison
+
+**Legacy Component:**
+
+<!-- Screenshot of legacy component will be added here -->
+
+**Spectrum 2 Component:**
+
+<!-- Screenshot of Spectrum 2 component will be added here -->
+
+### DOM structure changes
+
+<details>
+<summary>Spectrum Web Components:</summary>
+
+```html
+<sp-badge variant="informative" fixed="inline-start" size="m">
+    <sp-icon slot="icon" name="ui:CheckmarkSmall"></sp-icon>
+    <div class="label">
+        <slot></slot>
+    </div>
+</sp-badge>
+```
+
+</details>
+
+<details>
+<summary>Legacy (CSS main branch):</summary>
+
+```html
+<div
+    class="spectrum-Badge spectrum-Badge--sizeM spectrum-Badge--informative spectrum-Badge--fixed-inline-start"
+>
+    <svg
+        class="spectrum-Icon spectrum-Badge-icon"
+        focusable="false"
+        aria-hidden="true"
+    >
+        <!-- Icon content -->
+    </svg>
+    <div class="spectrum-Badge-label">Label text</div>
+</div>
+```
+
+</details>
+
+<details>
+<summary>Spectrum 2 (CSS spectrum-two branch):</summary>
+
+```html
+<div
+    class="spectrum-Badge spectrum-Badge--sizeM spectrum-Badge--informative spectrum-Badge--style-default spectrum-Badge--fixed-inline-start"
```

Comment:
I think this is the "default" class map: `spectrum-Badge spectrum-Badge--sizeM spectrum-Badge--neutral` instead.

looks like the prompt might be struggling with the class map variables (like when we don't add certain classes if an arg is a particular value)